### ISSUE TITLE
pubsub: subscriber stops after a minute

### ISSUE THREAD

**Issue by shouichi** [googleapis/google-cloud-ruby#4081]
I'm having the same issue as #2011. Subscriber stops after a minute without error (`Subscriber#on_error` is not called). Sending messages to the subscriber does not affect the behavior.

#### Environment details

  - OS: Linux
  - Ruby version: 2.6
  - Gem name and version: google-cloud-pubsub 0.39.2

#### Steps to reproduce

Start a subscriber and wait for a minute.

#### Code example

```ruby
start = Time.now

pubsub = Google::Cloud::PubSub.new(project_id: 'dummy')

topic = pubsub.create_topic "my-topic"

sub = topic.subscribe "my-sub"

subscriber = sub.listen do |received_message|
  received_message.acknowledge!
end

subscriber.on_error do |error|
  p error
end

subscriber.start

subscriber.wait!

p Time.now - time
# Reaches here after a minute.
```

#### Full backtrace

None.

**Comment by blowmage** [comment]
Hi @shouichi, thank you for the question. The behavior you are seeing is because [`Subscriber#wait!`](https://googleapis.dev/ruby/google-cloud-pubsub/latest/Google/Cloud/PubSub/Subscriber.html#wait!-instance_method) is not intended to be used before a call to [`Subscriber#stop`](https://googleapis.dev/ruby/google-cloud-pubsub/latest/Google/Cloud/PubSub/Subscriber.html#stop-instance_method) is made. In your case, the subscriber is still running after calling `subscriber.wait!`, because `subscriber.stop` was never called. If you call `subscriber.stopped?` afterwards it would return `false` because the subscriber is still running.

In other words, `Subscriber#wait!` is not intended to block a ruby process while the subscriber is running. To block indefinitely, I suggest calling `sleep` without passing an argument.

**Comment by blowmage** [comment]
I've looked into this some more, and while `Subscriber#wait!` is not intended to be used before a calling `Subscriber#stop`, it did work in releases before 0.39.2. So we will revert this and issue a new release.

**Comment by blowmage** [comment]
The fix is released in [v0.39.3](https://github.com/googleapis/google-cloud-ruby/releases/tag/google-cloud-pubsub/v0.39.3). Please upgrade to the latest. Thanks again!

**Comment by shouichi** [comment]
Confirmed the fix, thanks!
